notebook/agenteval_cq_math.ipynb

```diff
@@ -168,52 +112,39 @@
-    "\n",
-    "    Returns:\n",
-    "    - str: The log file without any information about the ground truth answer of the problem.\n",
-    "    \"\"\"\n",
-    "    f = open(file_name, \"r\").readlines()\n",
-    "    output_dictionary = \"\"\n",
-    "    for line in f:\n",
-    "        if \"is_correct\" not in line and \"correct_ans\" not in line and \"check_result\" not in line:\n",
-    "            output_dictionary += line\n",
-    "        elif \"is_correct\" in line:\n",
-    "            correctness = line.replace(\",\", \"\").split(\":\")[-1].rstrip().strip()\n",
-    "    return [output_dictionary, correctness]\n",
+    "def remove_ground_truth(test_case):\n",
+    "    test_details = json.loads(test_case)\n",
+    "    # need to remove the ground truth from the test details\n",
+    "    correctness = test_details.pop(\"is_correct\", None)\n",
+    "    test_details.pop(\"correct_ans\", None)\n",
+    "    test_details.pop(\"check_result\", None)\n",
+    "    return test_details, correctness\n",
     "\n",
     "\n",
     "# Reading one successful and one failed example of the task\n",
-    "response_successful = read_without_groundtruth(\n",
-    "    \"../test/test_files/agenteval-in-out/samples/sample_math_response_successful.txt\"\n",
-    ")[0]\n",
-    "response_failed = read_without_groundtruth(\n",
-    "    \"../test/test_files/agenteval-in-out/samples/sample_math_response_failed.txt\"\n",
-    ")[0]\n",
-    "\n",
-    "task = {\n",
-    "    \"name\": \"Math problem solving\",\n",
-    "    \"description\": \"Given any question, the system needs to  solve the problem as consisely and accurately as possible\",\n",
-    "    \"successful_response\": response_successful,\n",
-    "    \"failed_response\": response_failed,\n",
-    "}\n",
+    "success_str = open(\"../test/test_files/agenteval-in-out/samples/sample_math_response_successful.txt\", \"r\").read()\n",
+    "response_successful = remove_ground_truth(success_str)[0]\n",
+    "failed_str = open(\"../test/test_files/agenteval-in-out/samples/sample_math_response_failed.txt\", \"r\").read()\n",
+    "response_failed = remove_ground_truth(failed_str)[0]\n",
+    "\n",
+    "task = Task.parse_json_str(\n",
+    "    json.dumps(\n",
+    "        {\n",
+    "            \"name\": \"Math problem solving\",\n",
+    "            \"description\": \"Given any question, the system needs to solve the problem as consisely and accurately as possible\",\n",
+    "            \"successful_response\": response_successful,\n",
+    "            \"failed_response\": response_failed,\n",
+    "        }\n",
+    "    )\n",
+    ")\n",
     "\n",
-    "sys_msg = f\"\"\"Task: {task[\"name\"]}.\n",
-    "Task description: {task[\"description\"]}\n",
-    "Task successful example: {task[\"successful_response\"]}\n",
-    "Task failed example: {task[\"failed_response\"]}\n",
-    "\"\"\""
+    "criteria = generate_criteria(task=task, llm_config={\"config_list\": config_list})"
```

Comment:
I feel we should visualize the criteria for readers to understand the setup better.

Maybe `print(criteria.to_json())` here would help.